client/specs/App.test.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import App from '../src/pages/App/App';
+import userEvent from '@testing-library/user-event';
+import { BrowserRouter } from "react-router-dom";
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react';
+
+const response = {
+  data: {
+    user: {
+      name: 'John',
+      surname: 'Doe',
+    },
+    token: 'token123'
+  }
+};
+
+jest.mock('../src/hooks/useSignUp', () => ({
+  useSignUp: () => ({
+    signUp: jest.fn().mockResolvedValue(response),
+    loading: false,
+    error: null,
+  }),
+}));
+
+jest.mock('../src/hooks/useLogin', () => ({
+  useLogin: () => ({
+    login: jest.fn().mockResolvedValue(response),
+    loading: false,
+    error: null,
+  }),
+}));
+
+const mockNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockNavigate,
+}));
+
+
+describe('App', () => {
+  
+  describe('When SignUp', () => {
+    beforeEach(() => {
+      render(<BrowserRouter><App /></BrowserRouter>);
+      mockNavigate.mockClear();
+    });
+
+    test('should fail if any mandatory field is missing in form submission', async () => {
+      await act(async () => {
```

Comment:
I am not a huge fan of `act()` [here](https://medium.com/@AbbasPlusPlus/react-testing-library-understanding-act-and-when-to-use-it-301bd06fd1bc#:~:text=The%20Purpose%20of%20act(),in%20a%20real%20browser%20environment.) you can check a good article.
But this is ok!